### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by miguelfteixeira**
This PR brings a refresh on the documentation as well as a minor refactoring on the adquire method.

### CODE REVIEW

**Comment by JSFernandes** [comment]
:+1:

File: lib/cache_guard.rb [hunk 1]
```diff
@@ -29,9 +29,8 @@ def acquire
     # The FileStore increment operation returns nil if the cache key doesn't exist. Since the FileStore strategy
     # should be only used in development environment we can allow all guards to pass.
     # http://api.rubyonrails.org/classes/ActiveSupport/Cache/FileStore.html#method-i-increment
-    # http://www.rubydoc.info/github/mperham/dalli/ActiveSupport/Cache/DalliStore#increment-instance_method
     value = @cache_store.increment(@name, @default_value, :expires_in => @expires_in)
-    value.nil? || value == @default_value ? true : false
```
Comment (marionzualo) [inline lib/cache_guard.rb:34]: Doesn't this change require a version bump?

File: README.md [hunk 2]
```diff
@@ -40,12 +40,25 @@ cache_guard.guard { my_task_using_shared_resource }
...
+```ruby
+Rails.cache.increment(SecureRandom.uuid, 1)
+# => 1
+```
+
+We've successfully tested the following cache stores:
+* [Dalli][dalli_store_repository] v2.6.2
+* [Redis][redis_store_repository] v4.0.0
+
+You can also use [FileStore][file_store] on your development environment as long as you can afford that every guard
```
Comment (dsgh) [inline README.md]: This isn't very clear. Perhaps "afford" isn't the best word.

File: README.md [hunk 3]
```diff
@@ -4,22 +4,22 @@
...
-*cache_guard* is being used with *ActiveSupport::Cache::Store* 4.1 although it should be compatible with other versions.
-
 This gem is available on RubyGems:
 
+[https://rubygems.org/gems/cache_guard](https://rubygems.org/gems/cache_guard)
+
 ## Usage
-To use simply call the *guard* method passing a string that represents the name of the key that will be put in cache
-to block other guard calls and optionally an hash with options.
+To use simply call the *guard* method passing a string that represents the shared resource and optionally an
```
Comment (dsgh) [inline README.md]: a hash

File: README.md [hunk 4]
```diff
@@ -4,22 +4,22 @@
 
 ## Description
 
-*cache_guard* makes use of your application cache store to ensure that there is no parallel execution of a
-given block of code. You can think of *cache_guard* as a kind of *mutex* without the *wait* functionality.
+*cache_guard* makes use of your application [cache store][cache_stores] to ensure that there is no parallel execution
+of a given block of code. You can think of *cache_guard* as simple distributed *mutex* that instead of blocking raises
```
Comment (dsgh) [inline README.md]: as a simple

File: lib/cache_guard.rb (hunk 1 above)
Comment (miguelfteixeira) [inline lib/cache_guard.rb:34]: @marionzualo added a new commit with the version bump.

**Comment by dsgh** [comment]
:+1:

**Pull request merged**